### PR TITLE
[14.0][REM] documentation

### PR DESCRIPTION
**Pull request by legalsylvain**
Similar to : 
- https://github.com/OCA/OpenUpgrade/pull/3853
-  ~and https://github.com/OCA/OpenUpgrade/pull/3855~

but for v14.

### CODE REVIEW

File: .github/workflows/documentation-commit.yml [hunk 1]
```diff
@@ -1,44 +1,35 @@
...
+        uses: actions/setup-python@v4
         with:
-          python-version: 3.7
+          python-version: '3.10'
       - name: Check out Odoo
         uses: actions/checkout@v2
         with:
           repository: odoo/odoo
-          ref: "14.0"
+          ref: "16.0"
```
Comment (MiquelRForgeFlow) [inline .github/workflows/documentation-commit.yml]: 🤔

Comment (legalsylvain) [reply .github/workflows/documentation-commit.yml]: Great minds think alike ;-) See : https://github.com/OCA/OpenUpgrade/pull/3859#discussion_r1186923334

Answered in the other PR.

In the meantime, I removed the commit. Can be merged in my opinion.

Comment (MiquelRForgeFlow) [reply .github/workflows/documentation-commit.yml]: Yes, I saw later, that's why I resolved the comment.

**Comment by legalsylvain** [comment]
Hi @MiquelRForgeFlow, @pedrobaeza 

Agree to merge this simple one ? It's a good cleanup.

Note : once https://github.com/OCA/openupgradelib/pull/318 and https://github.com/OCA/OpenUpgrade/pull/3863 will be merged, the github workflow to generate openupgrade documentation will become very simple, and we can imagine to have a workflow on each "non-main" branch to update the documentation when a coverage file has changed.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**